src/compare_covermaps.py

```diff
@@ -512,55 +551,63 @@ def generate_report(dataset_name: str, country: str, true, pred) -> pd.DataFrame
         )""",
         resolution=10,
         crop_labels=[100],
+        years_covered=[2020],
     ),
     "glad": Covermap(
         "glad",
         'ee.ImageCollection("users/potapovpeter/Global_cropland_2019")',
         resolution=30,
         probability=0.5,
+        years_covered=[2003, 2007, 2011, 2015, 2019],
     ),
     # "gfsad": Covermap(
     #     "gfsad",
     #     ee.ImageCollection(ee.Image("USGS/GFSAD1000_V1")),
     #     resolution=1000,
     #     crop_labels=[1, 2, 3, 4, 5],
+    #     years_covered=[2010],
     # ),
     "asap": Covermap(
         "asap",
         'ee.ImageCollection(ee.Image("users/sbaber/asap_mask_crop_v03").unmask())',
         resolution=1000,
         probability=100,
         countries=[country for country in TEST_COUNTRIES.keys() if country != "Hawaii"],
+        years_covered=[2017],
     ),
     "dynamicworld": Covermap(
         "dynamicworld",
         """ee.ImageCollection(
             ee.ImageCollection("GOOGLE/DYNAMICWORLD/V1")
-            .filter(ee.Filter.date("2019-01-01", "2020-01-01"))
+            .filter(ee.Filter.date("2019-01-01", "2019-12-31"))
```

Comment:
Ah this filter date thing is a bit messy now with the years. What do you think about making it a regex style in all instances.
`.filterDate("YYYY-01-01", "YYYY-12-31")` and that YYYY is replaces with the year selected?
Also `filterDate` is the same as `.filter(ee.Filter.date` so it can be replaced everywhere

Reply:
I think that may work... the issue I had when thinking about this is that the Covermap objects get created before they are sampled. But as long as every dataset has a year specified then it should be fine.